test/gen-template/matter/external-addon-helper.js

```diff
@@ -0,0 +1,74 @@
+/**
+ *
+ *    Copyright (c) 2021 Silicon Labs
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+// This is an example of an external addon helper for templates.
+async function test_external_addon_helper() {
+  return 'This is example of test external addon helper.'
+}
+async function test_external_addon_all_events_helper(api) {
+  let events = await api.availableEvents(this)
+  let totalEvents = events.length
+  return totalEvents
+}
+async function test_external_addon_all_attributes_helper(api) {
+  let attributes = await api.availableAttributes(this)
+  let totalAttributes = attributes.length
+  return totalAttributes
+}
+async function test_external_addon_all_commands_helper(api) {
+  let commands = await api.availableCommands(this)
+  let totalCommands = commands.length
+  return totalCommands
+}
+async function test_external_addon_all_clusters_helper(api) {
```

Comment:
okay, I would keep "this" until someone said otherwise

Reply:
instead of context it gets "this". At first, I tried passing "context" but there are lots of helper functions and helpers of helper functions that are designed to deal with "this" so for now it seemed better to stick with it.

Moving helpers in general from "this" to context will require more work than just passing in as an argument

Reply:
I don't want to convert ALL the helpers to use context, just the "external" ones.
True, the Handlebars is using "this" heavily and I always found that use arcane, and it was always confusing what "this" was, like it always is with Javascript.

That's why I hoped that for the external helpers, instead of letting users guess what "this" is, we don't even bother them with it, since we are wrapping these helpers anyway.

But it's fine temporarily to use "this" and then before you hand this out to final users, change "this" into an explicit "context" argument.

As a general rule: if you can avoid making people use "this", avoid it.
But Handlebars is making you use "this", so for our internal helpers, we should continue using "this" and just be sad about it.

Reply:
ok changed from "this" to "context"